### PR TITLE
Clean riscv64 code, port Ld to Lw

### PR DESCRIPTION
**Pull request by VitalyAnkh**
I have cleaned some riscv64 code and port `Ld` related code (`IsLd`, `LdOffset`) to `Lw` (`IsLw`, `LwOffset`). There is still a debug check fail: line 4009 `DCHECK(assm_->IsLw(instr_lw))` in `src/codegen/riscv32/assembler-riscv32.cc` so it's WIP.

If you find something that could be improved, please let me know! Thanks!

### CODE REVIEW

**Comment by luyahan** [comment]
you should alter ld into lw in macro-assembler-riscv32.cc:1476
```
void TurboAssembler::li(Register rd, Operand j, LiFlags mode) {
  DCHECK(!j.is_reg());
  BlockTrampolinePoolScope block_trampoline_pool(this);
  if (!MustUseReg(j.rmode()) && mode == OPTIMIZE_SIZE) {
    UseScratchRegisterScope temps(this);
    int count = li_estimate(j.immediate(), temps.hasAvailable());
    int reverse_count = li_estimate(~j.immediate(), temps.hasAvailable());
    if (FLAG_riscv_constant_pool && count >= 4 && reverse_count >= 4) {
      // Lw a Address from a constant pool.
      RecordEntry((uint64_t)j.immediate(), j.rmode());
      auipc(rd, 0);
      // Record a value into constant pool.
      ld(rd, rd, 0);
    } else {
```

i suggest cleaned also  macro-assembler-riscv32.cc.

**Review by luyahan — CHANGES_REQUESTED** [review]
```
void ConstantPool::EmitPrologue(Alignment require_alignment) {
  // Recorded constant pool size is expressed in number of 32-bits words,
  // and includes prologue and alignment, but not the jump around the pool
  // and the size of the marker itself.
  const int marker_size = 1;
  int word_count =
      ComputeSize(Jump::kOmitted, require_alignment) / kInt32Size - marker_size;
  assm_->ld(zero_reg, zero_reg, word_count);
  assm_->EmitPoolGuard();
}
```

this ld also need to modify

**Comment by VitalyAnkh** [comment]
Thanks, will do it.

File: src/codegen/riscv32/constants-riscv32.h [hunk 1]
```diff
@@ -505,19 +482,6 @@ enum Opcode : uint32_t {
   RO_AMOMINU_W = AMO | (0b010 << kFunct3Shift) | (0b11000 << kFunct5Shift),
   RO_AMOMAXU_W = AMO | (0b010 << kFunct3Shift) | (0b11100 << kFunct5Shift),
 
-  // RV64A Standard Extension (in addition to RV32A)
-  RO_LR_D = AMO | (0b011 << kFunct3Shift) | (0b00010 << kFunct5Shift),
```
Comment (luyahan) [inline src/codegen/riscv32/constants-riscv32.h]: RV64A instr don't be deleted form assembler-riscv32.cc, so don't delete these.Let us open a new a pr to delete it.

File: src/codegen/riscv32/constants-riscv32.h [hunk 2]
```diff
@@ -547,12 +511,6 @@ enum Opcode : uint32_t {
   RO_FCVT_S_WU = OP_FP | (0b1101000 << kFunct7Shift) | (0b00001 << kRs2Shift),
   RO_FMV_W_X = OP_FP | (0b000 << kFunct3Shift) | (0b1111000 << kFunct7Shift),
 
-  // RV64F Standard Extension (in addition to RV32F)
```
Comment (luyahan) [inline src/codegen/riscv32/constants-riscv32.h]: here also

File: src/compiler/backend/riscv32/code-generator-riscv32.cc [hunk 3]
```diff
@@ -1576,6 +1576,7 @@ CodeGenerator::CodeGenResult CodeGenerator::AssembleArchInstruction(
     case kRiscvUlw:
       __ Ulw(i.OutputRegister(), i.MemoryOperand());
       break;
+<<<<<<< HEAD
```
Comment (luyahan) [inline src/compiler/backend/riscv32/code-generator-riscv32.cc]: these need delete

File: src/compiler/backend/riscv32/code-generator-riscv32.cc [hunk 4]
```diff
@@ -1585,15 +1586,14 @@ CodeGenerator::CodeGenResult CodeGenerator::AssembleArchInstruction(
     case kRiscvUld:
       __ Uld(i.OutputRegister(), i.MemoryOperand());
       break;
+=======
```
Comment (luyahan) [inline src/compiler/backend/riscv32/code-generator-riscv32.cc]: nedd delete

File: src/compiler/backend/riscv32/instruction-selector-riscv32.cc [hunk 5]
```diff
@@ -532,6 +531,9 @@ void InstructionSelector::VisitLoad(Node* node) {
     case MachineRepresentation::kTagged:         // Fall through.
       opcode = kRiscvLw;
       break;
+#endif
+    case MachineRepresentation::kWord64:
```
Comment (luyahan) [inline src/compiler/backend/riscv32/instruction-selector-riscv32.cc]: Line 544 have kWord64, so delete here

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 6]
```diff
@@ -911,24 +911,6 @@ void Decoder::DecodeRType(Instruction* instr) {
       Format(instr, "and       'rd, 'rs1, 'rs2");
       break;
 #ifdef V8_TARGET_ARCH_64_BIT
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: delete macro

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 7]
```diff
@@ -956,21 +938,6 @@ void Decoder::DecodeRType(Instruction* instr) {
       Format(instr, "remu      'rd, 'rs1, 'rs2");
       break;
 #ifdef V8_TARGET_ARCH_64_BIT
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: delete it

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 8]
```diff
@@ -1445,12 +1379,6 @@ void Decoder::DecodeIType(Instruction* instr) {
         Format(instr, "lhu       'rd, 'imm12('rs1)");
         break;
 #ifdef V8_TARGET_ARCH_64_BIT
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: delete it

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 9]
```diff
@@ -1498,23 +1426,6 @@ void Decoder::DecodeIType(Instruction* instr) {
         break;
       }
 #ifdef V8_TARGET_ARCH_64_BIT
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: delete it

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 10]
```diff
@@ -1655,9 +1566,6 @@ void Decoder::DecodeSType(Instruction* instr) {
         Format(instr, "sw        'rs2, 'offS('rs1)");
         break;
 #ifdef V8_TARGET_ARCH_64_BIT
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: delete it

File: src/diagnostics/riscv32/disasm-riscv32.cc [hunk 11]
```diff
@@ -1028,39 +995,6 @@ void Decoder::DecodeRAType(Instruction* instr) {
       Format(instr, "amomaxu.w'a 'rd, 'rs2, ('rs1)");
       break;
 #ifdef V8_TARGET_ARCH_64_BIT
-    case RO_LR_D:
```
Comment (luyahan) [inline src/diagnostics/riscv32/disasm-riscv32.cc]: Don't delete RV64A

File: test/unittests/compiler/riscv32/instruction-selector-riscv32-unittest.cc [hunk 12]
```diff
@@ -1008,8 +990,7 @@ static const MemoryAccess kMemoryAccesses[] = {
     {MachineType::Uint16(), kRiscvLhu, kRiscvSh},
     {MachineType::Int32(), kRiscvLw, kRiscvSw},
     {MachineType::Float32(), kRiscvLoadFloat, kRiscvStoreFloat},
-    {MachineType::Float64(), kRiscvLoadDouble, kRiscvStoreDouble},
```
Comment (luyahan) [inline test/unittests/compiler/riscv32/instruction-selector-riscv32-unittest.cc:1011]: Reserve float64, becaue rv32gc support double

File: test/unittests/compiler/riscv32/instruction-selector-riscv32-unittest.cc [hunk 13]
```diff
@@ -1558,46 +1526,6 @@ TEST_F(InstructionSelectorTest, Word32ReverseBytes) {
   }
 }
 
-TEST_F(InstructionSelectorTest, ExternalReferenceLoad1) {
```
Comment (luyahan) [inline test/unittests/compiler/riscv32/instruction-selector-riscv32-unittest.cc]: Resever these test. Open a new pr to fix it.
We can simply alter ld -> lw

**Review by luyahan — CHANGES_REQUESTED** [review]

File: src/codegen/riscv32/constants-riscv32.h (hunk 1 above)
Comment (VitalyAnkh) [reply src/codegen/riscv32/constants-riscv32.h]: How about deleting RV64A in this PR? I build it without errors, and d8 works as normal.

File: src/codegen/riscv32/constants-riscv32.h (hunk 2 above)
Comment (VitalyAnkh) [reply src/codegen/riscv32/constants-riscv32.h]: OK

**Review by luyahan — APPROVED** [review]

**Pull request merged**